### PR TITLE
Roles docs do not list built-in system roles

### PR DESCRIPTION
**Pull request by cwarnermm**
Documentation for: https://github.com/mattermost/docs/issues/4444

Updated:
- Self-Managed Admin Guide > Onboard Users > Advanced Permissions: Backend Infrastructure > Data Structure
   - Added a new subsection called **Built-In Roles** that lists all built-in roles and their permissions, as well as a link to the API documentation endpoint used to retrieve this information

### CODE REVIEW

File: source/deployment/permissions-backend.rst [hunk 1]
```diff
@@ -234,9 +234,372 @@ Roles are dynamic and user configurable, necessitating a database table with the
 - ``Description`` (String)
 - ``Permissions`` (String): Space-separated permissions names
 - ``Scheme Managed`` (bool): Indicates whether this role is managed as part of a scheme.
-- ``BuiltIn`` (bool): Indicates if this role is built in to the Mattermost system and not removable by the user.
-
-The System Scheme is built in to the product and it's roles are defined as ``BuiltIn: true`` in the ``Roles`` table.
+- ``BuiltIn`` (bool): Indicates if this role is built-in to the Mattermost system and not removable by the user.
+
+Built-In Roles
```
Comment (justinegeffen) [inline source/deployment/permissions-backend.rst:239]:
```suggestion
Built-in roles
```

File: source/deployment/permissions-backend.rst [hunk 2]
```diff
@@ -234,9 +234,372 @@ Roles are dynamic and user configurable, necessitating a database table with the
...
+~~~~~~~~~~~~~~
+
+The System Scheme is built-in to the product, and its roles are defined as ``BuiltIn: true`` in the ``Roles`` table. You can use the Mattermost API to `retreive a list of permissions by role name <https://api.mattermost.com/#tag/roles/paths/~1roles~1names/post>`__.
```
Comment (justinegeffen) [inline source/deployment/permissions-backend.rst:242]:
```suggestion
The System Scheme is built-in to the product, and its roles are defined as ``BuiltIn: true`` in the ``Roles`` table. You can use the Mattermost API to `retrieve a list of permissions by role name <https://api.mattermost.com/#tag/roles/paths/~1roles~1names/post>`__. 
```

File: source/deployment/permissions-backend.rst [hunk 3]
```diff
@@ -234,9 +234,372 @@ Roles are dynamic and user configurable, necessitating a database table with the
...
+- get_public_link
+- remove_reaction
+- manage_public_channel_members
+- use_slash_commands
+- upload_file
+- manage_private_channel_properties
+- create_post
+- edit_post
+
+
```
Comment (justinegeffen) [inline source/deployment/permissions-backend.rst:292]:
```suggestion
```

**Review by justinegeffen — COMMENTED** [review]
Minor suggestions added.

Question: Should the role permissions have backticks around them to indicate that they're part of the system code (for want of a better term)? I am 0/5 on it and I don't believe we have these roles mentioned elsewhere so there's no specific reference I have in mind.

**Comment by cwarnermm** [comment]
It's a good question, @justinegeffen. Given that we're talking about the contents of database tables, I'm inclined to use formatting that suggests system code (like backticks).

**Review by adamjclarkson — APPROVED** [review]

**Review by icelander — APPROVED** [review]
LGTM!

**Review by justinegeffen — APPROVED** [review]

**Pull request merged**